File: cpp/2633.cpp
#include <iostream>
#include <string>
#include <map>

using namespace std;

int main()
{
    int n;
    
    while (cin >> n) {
        map<int, string> exp_map;
        string meat, result = "";
        int expiration;

        for (n; n > 0; n--) {
            cin >> meat >> expiration;
            exp_map[expiration] = meat;
        }

        for (auto i: exp_map) {
            result += i.second + ' ';
        }

        result.pop_back();
        cout << result << '\n';
    }

    return 0;
}

File: py/2633.py
def main():
    while True:
        try:
            n = int(input())
            map = dict()

            for _ in range(n):
                meat, expiration = input().split()
                expiration = int(expiration)
                map[expiration] = meat

            result = [i[1] for i in sorted(map.items())]
            print(*result)

        except EOFError:
            break


main()